### ISSUE TITLE
systemd tor.service failed to start on Fedora

### ISSUE THREAD

**Issue by bmeneguele** [GouveaHeitor/nipe#83]
Distro: Fedora 31
Init: systemd 243 (v243.5-1.fc31)
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified
Perl: v5.30.1

I'm still not sure the reason for such error:

```
$ ./nipe.pl start
Job for tor.service failed because the control process exited with error code.
See "systemctl status tor.service" and "journalctl -xe" for details.
$ journalctl -xe
(...)
Jan 16 17:04:17 glitch systemd[1]: tor.service: Start request repeated too quickly.
Jan 16 17:04:17 glitch systemd[1]: tor.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- The unit tor.service has entered the 'failed' state with result 'exit-code'.
Jan 16 17:04:17 glitch systemd[1]: Failed to start Anonymizing overlay network for TCP.
-- Subject: A start job for unit tor.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- A start job for unit tor.service has finished with a failure.
-- 
-- The job identifier is 166485 and the job result is failed.
```

I'm going to investigate it further.
@GouveaHeitor you can assign this issue to me.

**Comment by bmeneguele** [comment]
Ok, just as a note: we might have some issue with our custom tor configs:
```
Jan 16 17:51:01.561 [warn] Error creating directory /var/lib/tor: Permission denied
Jan 16 17:51:01.561 [warn] Failed to parse/validate config: Couldn't create private data directory "/var/lib/tor"
```
Will keep digging on that.

**Comment by bmeneguele** [comment]
Ok, the issue with that were two:
1. the "User" key on our custom tor config should be respected and not changed, lol. I just realized, looking at how the 'tor' package is installed in Fedora (the distro I use) and the User 'toranon' was actually set by the packagers instead of the UID 19, which upstream was forcing. I'm not sure the policy around that, but will try to check it later. But in short, just don't change the "User" key in config file, since DataDirectory is created using this user's uid:gid.

2. Fedora has being more restrictive every release regarding user's permission by adopting a "block all, allow as necessary" approach with SELinux. Because of that it was necessary to unblock dns_port setting for the user:
`# semanage port -a -t dns_port_t -p udp`
in order to get it running correctly.

After that I could get Nipe up and running.
As a conclusion, I think that both points are worth to add a note either in source code and README file, just to make sure no one else get stuck like I did.

I'm going to send a patch for README adding these info.
